src/mgr/InfoAction.h

```diff
@@ -28,67 +28,67 @@ class InfoActionData
 public:
     struct timeval squid_start;
```

Comment:
If we can use [zero-initialization](https://en.cppreference.com/w/cpp/language/zero_initialization) instead of memset(), we should:

```suggestion
     struct timeval squid_start = {};
```

I do not know whether we can use zero-initialization here (and in a few similar places in this PR), but I hope we can. Please check if you have not already.

Reply:
Good catch, thanks.

I'm not sure what you mean by "a few other places". There is only one remaining use of memset() and that is for a char array, not a struct.

Reply:
> There is only one remaining use of memset() and that is for a char array, not a struct. Doing zero-init there results in "error: char-array initialized from wide string".

Strange. This works in my tests:

```
#include <iostream>

class Foo {
public:
    char array1[35] = {};
};

int main()
{
  Foo foo;
  std::cout << (int)foo.array1[0] << (int)foo.array1[1] << (int)foo.array1[2] <<  (int)foo.array1[3] << "\n";
}
```

Reply:
Found the issue. I was trying the older CharT form of zero-initialization (key[35] = "") which is supposedly more supported than the newer C++11 initialization form(s).

But that empty string is apparently modified by locale into implicit "wide string" chars. GCC-7 and GCC-8 produced that error mentionend earlier, GCC-9 was happy either way.

Using the {} zero-initialization does seem to work okay in all of them for the small test you provided. So going with that provided it passes the full in-situ build checks.

Reply:
Glad you figured it out.